infrastructure/build/fab/lfric_base.py

```diff
@@ -48,7 +46,16 @@ class LFRicBase(FabBase):
     def __init__(self, name: str,
                  root_symbol: Optional[Union[List[str], str]] = None):
 
+        # List of all precision preprocessor symbols and their default.
+        # Used to add corresponding command line options, and then to define
+        # the preprocessor definitions.
+        self._all_precisions = [("RDEF_PRECISION", "64"),
+                                ("R_SOLVER_PRECISION", "32"),
+                                ("R_TRAN_PRECISION", "64"),
+                                ("R_BL_PRECISION", "64")]
+
         super().__init__(name)
+
```

Comment:
Line 76:
•	The whole business for finding the applications source is byzantine and, again, fragile. It's also in the wrong place. Core, and that includes core's build system, shouldn't know anything about the applications repository. It is a general infrastructure for earth system modelling, there's no requirement it be used with the LFRic applications. As such there should be an extension of {{{LFRicBase}}} called, maybe,  {{{ApplicationsBase}}} which is defined in the application repository. This removes the need for finding the application working copy since it will be in it.

Reply:
That is I believe part of a wider discussion surrounding the build.sh script we use. I'll start a special comment about this later, and get everyone else involved for feedback.